src/generators/dom/index.js

```diff
@@ -54,19 +54,38 @@ class DomGenerator extends Generator {
 
 		if ( fragment.key ) properties.addBlock( `key: key,` );
 
-		properties.addBlock( deindent`
-			mount: function ( target, anchor ) {
-				${fragment.builders.mount}
-			},
+		if ( fragment.builders.mount.isEmpty() ) {
+			this.uses.noop = true;
+			properties.addBlock( `mount: noop,` );
+		} else {
+			properties.addBlock( deindent`
+				mount: function ( target, anchor ) {
+					${fragment.builders.mount}
+				},
+			` );
+		}
 
-			update: function ( changed, ${fragment.params} ) {
-				${fragment.builders.update}
-			},
+		if ( fragment.builders.update.isEmpty() ) {
```

Comment:
`update` seems sensible, but can this case actually happen for `mount` and `teardown`?